### ISSUE TITLE
Allow swapping in community-finetuned VAEs (start with ScragVAE)

### ISSUE THREAD

**Issue by ChuxiJ** [ace-step/ACE-Step-1.5#1138]
## Summary

Add first-class support for swapping the default Oobleck VAE with a
community-finetuned alternative — starting with
[**ScragVAE**](https://huggingface.co/scragnog/Ace-Step-1.5-ScragVAE),
a fine-tune of `ACE-Step/ace-step-v1.5-1d-vae-stable-audio-format` by
@scragnog.

ScragVAE ships the same `AutoencoderOobleck` architecture (`config.json`
+ `diffusion_pytorch_model.safetensors`), so it can be loaded by the
existing `_load_vae_model` path with zero architectural changes — what
we lack today is a way to **point at a different checkpoint directory
without editing source**.

## Motivation

- Several users have asked for a way to A/B alternative VAEs against
  the bundled one (clarity, transient handling, vocal grain).
- The cost is small: a registry entry + a path override + one Gradio
  dropdown. The MLX path comes "for free" because
  `MLXAutoEncoderOobleck.from_pytorch_config(self.vae)` builds itself
  from whatever PyTorch VAE we loaded.
- Community fine-tunes are a healthy ecosystem signal; making them
  swappable lowers the barrier for more.

## Proposed design

Three layers, mirroring how DiT variants are already handled:

### 1. Downloader registry (`acestep/model_downloader.py`)

Add a sibling to `SUBMODEL_REGISTRY` for VAE variants:

```python
VAE_REGISTRY: dict[str, str] = {
    # variant_id -> HF repo_id ; subdir under <checkpoints>/ uses variant_id
    "scragvae": "scragnog/Ace-Step-1.5-ScragVAE",
}
```

Plus a `download_vae(variant, checkpoints_dir, prefer_source)` helper
that re-uses the existing `_smart_download` (HF → ModelScope fallback).

The bundled official VAE keeps its current location:
`<checkpoints>/vae/` — no migration needed.

### 2. VAE path resolution

Today, `init_service_loader_components.py:40` hardcodes
`vae_checkpoint_path = os.path.join(checkpoint_dir, "vae")`.

Change to a small resolver:

```python
def _resolve_vae_path(checkpoint_dir, vae_variant):
    """vae_variant may be 'official', a key in VAE_REGISTRY, or an absolute path."""
    if vae_variant and os.path.isabs(vae_variant) and os.path.isdir(vae_variant):
        return vae_variant
    if not vae_variant or vae_variant == "official":
        return os.path.join(checkpoint_dir, "vae")
    if vae_variant in VAE_REGISTRY:
        return os.path.join(checkpoint_dir, vae_variant)
    raise ValueError(f"Unknown VAE variant: {vae_variant!r}")
```

Resolution order, in `initialize_service`:

1. Explicit `vae_checkpoint` parameter (passed in by Gradio / API)
2. `ACESTEP_VAE_CHECKPOINT` env var (variant id or absolute path)
3. Default `"official"` → existing `<checkpoints>/vae/`

If the resolved variant is in `VAE_REGISTRY` and the directory is
missing, auto-download via `download_vae(...)` — same UX as DiT
variants today.

### 3. Gradio UI

Add a single `Dropdown` next to the model-variant selector in
`acestep/ui/gradio/interfaces/generation_service_config_rows.py`:

- Label: **"VAE"**
- Choices: `["official", "scragvae"]` (auto-extended from `VAE_REGISTRY`)
- Default: `"official"`
- Wired through `service_init.py` into the existing `initialize_service`
  call so reloading the model swaps the VAE in-place.

### MLX path

No changes needed. `_init_mlx_vae` builds its MLX state from
`self.vae` (the loaded torch model), so it transparently picks up
whichever VAE we just loaded.

## Out of scope (for the first PR)

- Quantized GGUF loading (`scragvae-BF16.gguf`) — separate feature.
- Per-track VAE override at generate time (would require two VAEs in
  memory).
- `training_v2/model_loader.py` integration (training pipelines stay on
  the bundled VAE for now).

## Files touched (preview)

- `acestep/model_downloader.py` — `VAE_REGISTRY`, `download_vae()`
- `acestep/core/generation/handler/init_service_loader_components.py` —
  `_resolve_vae_path`, `_load_vae_model` signature
- `acestep/core/generation/handler/init_service_orchestrator.py` —
  thread `vae_checkpoint` through `initialize_service`
- `acestep/ui/gradio/interfaces/generation_service_config_rows.py` — VAE dropdown
- `acestep/ui/gradio/events/generation/service_init.py` — pass dropdown value
- `docs/en/ALT_VAE.md` — short user-facing doc, with credit + license note
- Tests for path resolution (env var / param / default fall-through)

## Credits & license

ScragVAE is MIT-licensed (per the model card) and credits ACE-Step as
the base model. We'll surface that in `docs/en/ALT_VAE.md` and in the
PR description; no weights are vendored.

---

I'll send a PR shortly. Filing this issue first so the design is
discussable and so the PR can `Closes #N`.
